### ISSUE TITLE
Address encoding of block.coinbase_account

### ISSUE THREAD

**Issue by graup** [aergoio/herapy#70]
`block.coinbase_account` should be a b58check encoded address.

Current result: `eZDSP5tk3jz49yjxiF2XMkfou3VQtDRxVow1PNPuz3Qe`

Expected result: ` AmLrV7tg69KE5ZQehkjGiA7yJyDxJ25uyF96PMRptfzwozhAJbaK`

https://github.com/aergoio/herapy/blob/92845d5a601fc86eebc180522a2b9898ee266f67/aergo/herapy/obj/block.py#L203

**Comment by hanlsin** [comment]
Hi @graup,

Sorry for late response.

The coinbase account is printing `base58.Encode`, not `base58check`.

https://github.com/aergoio/aergo/blob/35879e34e615f2498a31381ffa97160e98211379/cmd/aergocli/util/base58addr.go#L243

Where did you find to use `base58check` for the coinbase account?

**Comment by graup** [comment]
@hanlsin Exactly, and that is the problem. It SHOULD be base58check, because it is an address.

**Comment by hanlsin** [comment]
In that case, it's not only base58check, but also `AddressVersion + base58check(addr)`. Right?

I will report it to Aergo CLI and make it right.

**Comment by graup** [comment]
More like `base58check(addr, AddressVersion)`, but yes. Thanks

**Comment by graup** [comment]
Cant you just replace the line with `types.EncodeAddress(self.coinbase_account)`?

**Comment by graup** [comment]
Sorry, I misunderstood your previous message. I didn't realize this bug (if I'm right) was also in aergocli. Let's fix it there first!

**Comment by hanlsin** [comment]
I reported it for aergocli as a question at first, because I don't know the decision about it.

https://github.com/aergoio/aergo/issues/91

And, you are right. Just, in aergocli, replacing

https://github.com/aergoio/aergo/blob/35879e34e615f2498a31381ffa97160e98211379/cmd/aergocli/util/base58addr.go#L243

to
```go
		out.Header.CoinbaseAccount = types.EncodeAddress(b.GetHeader().GetCoinbaseAccount())
```

is the simple answer.

**Comment by hanlsin** [comment]
> More like `base58check(addr, AddressVersion)`, but yes. Thanks

Yes. If I may allow to fix it little bit, then it would be `base58check(AddressVersion, addr)`, because the version is in front of the address. :)

**Comment by hanlsin** [comment]
https://github.com/aergoio/aergo/issues/91

In the issue above, the only `base58` expression was mistake. So I will modify herapy for the next release too.

**Comment by hanlsin** [comment]
It is included in version 1.3.3.
